src/extension.ts

```diff
@@ -175,7 +175,16 @@ export async function activate(context: vscode.ExtensionContext) {
 
 		vscode.commands.registerCommand('camelk.integrations.createNewIntegrationFile', async (...args:any[]) => { await NewIntegrationFileCommand.create(args);});
 		vscode.commands.registerCommand('camelk.integrations.selectFirstNode', () => { selectFirstItemInTree();});
-		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => { await downloadSpecificCamelKJavaDependencies(context, uri, mainOutputChannel)});
+		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => {
+			vscode.window.withProgress({
+				location: vscode.ProgressLocation.Notification,
+				title: "Updating the classpath...",
+				cancellable: false
+			}, async (progress) => {
+				progress.report({ increment: -1 });
+				return downloadSpecificCamelKJavaDependencies(context, uri, mainOutputChannel);
```

Comment:
It is reporting only for a subpart of the refresh.
There are 2 parts for the user to really benefits from the classpath refresh:
- the update of the folder containing all dependencies that will be part of the classpath
- the Java Language Server reading the folder and updating the classpath and recompiling.

We are not waiting here that the Java Language Server is picking the new files

Reply:
That seems to be an issue of the underlying functionality then. I am waiting until the command promise is resolved. If that is not enough, then there seems to be a logical issue in how we call / trigger the java lsp cp refresh as it seems we are not waiting for it to finish before resolving the command.

Reply:
Ok, we are basically lacking a handle on the Java LSP classpath update and hence can't wait for it to be done in a nice way. I am going to rename the text of the progress popup so it becomes more obvious we are only covering the Camel K dependencies update. This needs to be revisited once Java LSP provides an API to call the refresh.